server/admin.api.js

```diff
@@ -138,6 +139,150 @@ router.get('/admin/stats', mw.adminOnly, async (req, res)=>{
 	}
 });
 
+router.get('/api/lock/count', mw.adminOnly, async (req, res)=>{
+	try {
+		const countLocksQuery = {
+			lock : { $exists: true }
+		};
+		const count = await HomebrewModel.countDocuments(countLocksQuery)
+			.then((result)=>{
+				return result;
+			});
+		return res.json({
+			count
+		});
+	} catch (error) {
+		console.error(error);
+		return res.json({ status: 'ERROR', detail: 'Unable to get lock count', error });
+	}
+});
+
+router.post('/api/lock/:id', mw.adminOnly, async (req, res)=>{
+	try {
+		const lock = req.body;
+		lock.applied = new Date;
+
+		const filter = {
+			shareId : req.params.id
+		};
+
+		const brew = await HomebrewModel.findOne(filter);
+
+		if(brew.lock) {
+			// console.log('ALREADY LOCKED');
+			return res.json({ status: 'ALREADY LOCKED', detail: `Lock already exists on brew ${req.params.id} - ${brew.title}` });
+		}
+
+		brew.lock = lock;
+		brew.markModified('lock');
+
+		await brew.save();
+
+		// console.log(`Lock applied to brew ID ${brew.shareId} - ${brew.title}`);
+		return res.json({ status: 'LOCKED', detail: `Lock applied to brew ID ${brew.shareId} - ${brew.title}`, ...lock });
+	} catch (error) {
+		console.error(error);
+		return res.json({ status: 'ERROR', error, message: `Unable to set lock on brew ${req.params.id}` });
```

Comment:
We may want to have the errors utilize our central error handling middleware. I.e., `throw` the error here, and the error handler in `app.js` will organize the error into a common format (include a stack trace, etc.) and send it.